compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/util/ImmutableTypeCloner.java

```diff
@@ -230,7 +230,9 @@ private static BIntersectionType setImmutableType(Location pos, Types types,
                                                                           pkgId, owner, symTable, anonymousModelHelper,
                                                                           names, unresolvedTypes),
                                                 immutableXmlTSymbol, origXmlType.flags | Flags.READONLY);
-                immutableXmlTSymbol.type = immutableXmlType;
+                if (immutableXmlTSymbol != null) {
```

Comment:
Why can this be null now?

Reply:
I am getting NPE here for some scenarios. Is that a bug? If so, shall I remove this check and comment the failing tests?  (This check is already used for some other types. That's why I added for XML case also)

Reply:
We shouldn't disable tests. Is this a new test related to your changes? If so an XML type may be getting created incorrectly, we can check and fix that.

Reply:
> Is this a new test related to your changes

Yes

Reply:
Created an issue for this. https://github.com/ballerina-platform/ballerina-lang/issues/36730